subprojects/model-core/src/main/java/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStore.java

```diff
@@ -320,22 +334,50 @@ private PropertyAnnotationMetadataBuilder getOrCreateBuilder(String propertyName
                 .forEach(entry -> {
                     String fieldName = entry.getKey();
                     ImmutableMap<Class<? extends Annotation>, Annotation> fieldAnnotations = entry.getValue();
-                    validationContext.visitTypeProblem(WARNING,
-                        type,
-                        String.format("field '%s' without corresponding getter has been annotated with %s",
-                            fieldName,
-                            simpleAnnotationNames(fieldAnnotations.keySet().stream()))
+                    validationContext.visitTypeProblem(problem ->
+                        problem.withId(ValidationProblemId.IGNORED_ANNOTATIONS_ON_FIELD)
+                            .forType(type)
+                            .reportAs(ERROR)
+                            .withDescription(() -> String.format("field '%s' without corresponding getter has been annotated with %s", fieldName, simpleAnnotationNames(fieldAnnotations.keySet().stream())))
+                            .happensBecause("Annotations on fields are only used if there's a corresponding getter for the field")
+                            .withLongDescription("If a field is annotated but that there's no corresponding getter, then the annotations are ignored")
+                            .addPossibleSolution(() -> "Add a getter for field '" + fieldName + "'")
+                            .addPossibleSolution(() -> "Remove the annotations on '" + fieldName + "'")
+                            .documentedAt("validation_problems", "ignored_annotations_on_field")
                     );
                 });
         }
         return propertiesMetadata;
     }
 
-    private void processMethodAnnotations(Method method, Map<String, PropertyAnnotationMetadataBuilder> methodBuilders, TypeValidationContext validationContext) {
+    private boolean shouldIgnore(Method method) {
         if (method.isSynthetic()) {
-            return;
+            return true;
         }
         if (method.isBridge()) {
+            return true;
+        }
+        if (isIgnoredGeneratedGroovyMethod(method)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Groovy getters are annotated with `@Generated` but shouldn't be ignored.
+     * Other methods which have '$' in their name are usually a good heuristic for the
+     * fact we should ignore them, for example methods generated by the @Memoized annotation
+     * ref: https://github.com/gradle/gradle/issues/16117
+     *
+     * @param method a candidate method
+     * @return true if we suspect this method to be a safely ignorable generated method
+     */
+    private boolean isIgnoredGeneratedGroovyMethod(Method method) {
```

Comment:
We need to add a test to show that the `@Memoized` method is actually ignored. The smoke test is not enough.